### ISSUE TITLE
Unable to get password in Archer C1200 V2.0

### ISSUE THREAD

**Issue by dark-N00B** [AlexandrErohin/home-assistant-tplink-router#103]
**Describe the bug**
![image](https://github.com/user-attachments/assets/22c2566f-7b15-4d30-865e-4c819889c24b)

**Additional Information (please complete the following information)**
 - Router Hardware Version: Archer C1200 v2.0
 - Router Firmware Version: 2.0.3 Build 20211219 Rel. 67401 (EU)
 - Home Assistant Version: 2024.10.3
 - TP-Link Router Integration Component Version: v1.18.0

**Comment by AlexandrErohin** [comment]
@dark-N00B Have you typed in the password you use to login into the password field on login page, but without  pressing Enter to login to the router?

**Comment by dark-N00B** [comment]
> @dark-N00B Have you typed in the password you use to login into the password field on login page, but without pressing Enter to login to the router?

Yes, I first tried with saved password, later on tried typing it and then get the element, in both cases it failed.

**Comment by AlexandrErohin** [comment]
You may try to contact @swwgames he has added C1200 V2 support here https://github.com/AlexandrErohin/TP-Link-Archer-C6U/pull/10

**Comment by dark-N00B** [comment]
@swwgames How did you manage to get the value? I'm getting an empty string!

**Comment by swwgames** [comment]
Hi @dark-N00B,

I checked my old pull request. And in there I described the following steps:

1. Go to the webinterface of your TP-link router and open the developertools(ctrl+shift+i or F12). 
 2. In the developertools, navigate to the network tab. 
 3. Make sure the recoder is enabeld. 
 4. Login to your wifi router 
 5. There should appear activity in the networking tab. 
 6. Scroll up until you find "login?form=login" and click it. 
 7. Click payload and copy the string of characters. It should be a long string. 
 8. This is the web encrypted password.

Let me know if this works or if you need a more detailed explanation.

**Comment by swwgames** [comment]
That is odd, It looks like your router uses a completely different payload to log in. Maybe your router is set up differently? (I use mine as an accesspoint) You could try a factory reset, other wise you can try and add support for this different kind of login. Maybe @AlexandrErohin, do you recognize this payload?

**Comment by swwgames** [comment]
Hi, today I worked on an encryption method for the Archer C1200. 
So now I do recognize the response from the router you send me. This response is used in the encryption process. 
If all goes well, you won’t need to use the web-encrypted password in the near future.

Also, there should be another `login?form=login` request in the logs, there in the payload should be the password.
![image](https://github.com/user-attachments/assets/a3f449f0-6637-4116-a0e7-0a5c705dca5c)
This image shows how the payload looks on my machine for refrence.

Let me know if you can find the web encrypted password, and I will keep you posted on my progress.

**Comment by AlexandrErohin** [comment]
@dark-N00B Do you have the latest Router Firmware Version for your router?
@swwgames Good news! Would be great if you could add it to the client!

**Comment by dark-N00B** [comment]
> @dark-N00B Do you have the latest Router Firmware Version for your router?

Yes, the firmware version is latest. The above-mentioned version is the same as this [link](https://www.tp-link.com/en/support/download/archer-c1200/v2/#Firmware)

**Comment by dark-N00B** [comment]
@swwgames & @AlexandrErohin 
I had to set network configurator as offline to catch this:
![image](https://github.com/user-attachments/assets/bb7db57b-2dba-41d2-adf9-955e8faa9154)

But using this as password still has issues
![image](https://github.com/user-attachments/assets/31950aa1-8f59-477e-977a-caaf0d5be7ab)

**Comment by dark-N00B** [comment]
> Hi, today I worked on an encryption method for the Archer C1200. So now I do recognize the response from the router you send me. This response is used in the encryption process. If all goes well, you won’t need to use the web-encrypted password in the near future.

Hoping for this as this will get a lot easier

**Comment by AlexandrErohin** [comment]
@dark-N00B You have set login via TP-LINK ID. This integration works ONLY with Local Password which is for Log In with Local Password https://raw.githubusercontent.com/AlexandrErohin/home-assistant-tplink-router/master/docs/media/router.png

**Comment by AlexandrErohin** [comment]
@dark-N00B Any news?
You may also try to install version 2.0.0 - in this version you should pass the plain password that you pass to the router without any modifications.

**Comment by dark-N00B** [comment]
> Any news?

Thanks for asking, but I unbound my router from TP-Link cloud without backing up my router config, now I don't remember my local password as I set it up almost 6 years ago. I'm trying to find a better time to reset my router and configure all those DHCP IP-MAC bindings manually later.

> You may also try to install version 2.0.0

And, this is a very good news, good work in finding out the encryption process

**Comment by dark-N00B** [comment]
Later on I was able to reset my router and login after I switched back to local password.

Thanks to all the effort and help.

Hope the TP-Link id login comes soon too!!
